docs/how-tos/hello-world-module.md

```diff
@@ -0,0 +1,304 @@
+
+## Choose an API to implement
+
+Now it's time to decide which standard Viam [API](/appendix/apis/) makes the most sense for your module.
+You need a way to return an image, and you need a way to return a number.
+
+If you look at the [camera API](/appendix/apis/components/camera/), you can see the `GetImage` method, which returns an image.
+This seems promising.
+None of the camera API methods return a number though.
+
+Look at the [sensor API](/appendix/apis/components/sensor/), which includes the `GetReadings` method.
+You could return a number with that, but the sensor API can't return an image.
+
+Your module can support multiple modular resources, so for our purposes here, let's make two modular resources: a camera to return the image, and a sensor to return a random number.
+
+## Generate stub files
+
+The easiest way to generate the necessary files for your module is to use the [Viam CLI](/cli/).
+
+1. Run the `module generate` command in your terminal:
+
+    ```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+    viam module generate
+    ```
+
+1. Follow the prompts, selecting the following options:
+
+   - Module name: `hello-world`
+   - Language: Your choice
+   - Visibility: `Private`
+   - Namespace/Organization ID:
+     - In the [Viam app](https://app.viam.com), navigate to your organization settings through the menu in upper right corner of the page.
+       Find the **Public namespace** and copy that string.
+   - Resource to add to the module (API): `Camera Component`.
+   We will add the sensor later.
+   - Model name: `hello-camera`
+   - Enable cloud build: `No`
+   - Register module: `No`
+
+1. Hit your Enter key and the generator will generate a folder called <file>hello-world</file> containing stub files for your modular camera.
+
+## Implement the API
+
+Edit the stub files to implement your test script in a way that works with the camera API:
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+1. Open the <file>hello-world/src/main.py</file> file in your code editing program.
+
+1. Add the following to the list of imports at the top of <file>main.py</file>:
+
+    ```python {class="line-numbers linkable-line-numbers"}
+    from viam.media.utils.pil import pil_to_viam_image
+    from viam.media.video import CameraMimeType
+    from viam.utils import struct_to_dict
+    from PIL import Image
+    ```
+
+1. Instead of hard-coding the path to the image into the module itself, make the path a configurable attribute so you or other users of the module can easily set a different path.
+   Add the following lines to the `reconfigure()` function definition:
+
+    ```python {class="line-numbers"}
+    attrs = struct_to_dict(config.attributes)
+    self.image_path = str(attrs.get("image_path"))
+    ```
+
+1. Since the camera won't work unless the user configures an `image_path` attribute, add the following code to the `validate()` function to throw an error if `image_path` isn't configured:
```

Comment:
```suggestion
1. Since the camera must know which file to show, `image_path` is a required attribute. That means someone using the module has to configure an `image_path`. Add the following code to the `validate()` function to throw an error if `image_path` isn't configured:
```

Reply:
Am I allowed to anthropomorphize the camera like that?
```suggestion
1. Since the camera needs to get an image from somewhere, `image_path` is a required attribute.
   That means someone using the module must configure an `image_path`.
   Add the following code to the `validate()` function to throw an error if `image_path` isn't configured:
```